src/main/java/net/snowflake/ingest/streaming/internal/AbstractRowBuffer.java

```diff
@@ -463,4 +481,43 @@ static EpInfo buildEpInfoFromStats(long rowCount, Map<String, RowBufferStats> co
     }
     return epInfo;
   }
+
+  /** Row buffer factory. */
+  static <T> AbstractRowBuffer<T> createRowBuffer(
+      BufferInitialisationContext initialisationContext, ChannelState channelState) {
+    switch (initialisationContext.bdecVersion) {
+      case ONE:
+        //noinspection unchecked
+        return (AbstractRowBuffer<T>) new ArrowRowBuffer(initialisationContext, channelState);
+      case THREE:
+        //noinspection unchecked
+        return (AbstractRowBuffer<T>) new ParquetRowBuffer(initialisationContext, channelState);
+      default:
+        throw new SFException(
+            ErrorCode.INTERNAL_ERROR,
+            "Unsupported BDEC format version: " + initialisationContext.bdecVersion);
+    }
+  }
+
+  /** Initialisation context for the buffer. */
+  static class BufferInitialisationContext {
+    final OpenChannelRequest.OnErrorOption onErrorOption;
+    final BufferAllocator allocator;
+    final Constants.BdecVersion bdecVersion;
+    final String fullyQualifiedChannelName;
+    final Consumer<Float> rowSizeMetric;
+
+    BufferInitialisationContext(
+        OpenChannelRequest.OnErrorOption onErrorOption,
+        BufferAllocator allocator,
+        Constants.BdecVersion bdecVersion,
+        String fullyQualifiedChannelName,
+        Consumer<Float> rowSizeMetric) {
+      this.onErrorOption = onErrorOption;
+      this.allocator = allocator;
+      this.bdecVersion = bdecVersion;
```

Comment:
Looks like this is not used?

Reply:
It is used to decide what buffer to create, I can pull it out to the method